### ISSUE TITLE
Stuck on which loss function to force all samples of once class together 

### ISSUE THREAD

**Issue by jlim13** [KevinMusgrave/pytorch-metric-learning#244]
This is a really cool library, thanks for open sourcing this! 
I read through this post https://github.com/KevinMusgrave/pytorch-metric-learning/issues/6 and it seems to be kinda similar to what I am looking for. 

Let's say I have a batch of `N` samples where I have `n_a` elements in class `A` and `n_b` elements in class `B`, where `n_a + n_b = N`. I also have a list of `N` labels, `L`, indicating if a given sample is in `A` or `B`. Next, I compute a cosine similarity matrix that is of shape `NxN`. Which loss function do you think I should look at if I want to force all elements of `A` and `B` to be in their own clusters for a given batch. Thanks!

**Comment by KevinMusgrave** [comment]
All the loss functions are meant to cluster your embeddings. NTXentLoss is widely used, so you can start with that:

```python
from pytorch_metric_learning.losses import NTXentLoss
loss_fn = NTXentLoss()

# cosine similarity matrix is computed in the loss function
loss = loss_fn(embeddings, L)
```

**Comment by jlim13** [comment]
Thanks for the prompt response and insight! So to clarify, embeddings has shape `Batch x Embedding Dimension`?

**Comment by KevinMusgrave** [comment]
Yes

**Comment by jlim13** [comment]
Sweet, thanks!
